pkg/maven/settings.go

```diff
@@ -0,0 +1,150 @@
+package maven
+
+import (
+	"errors"
+	"fmt"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var getenv = os.Getenv
+
+// SettingsFileType ...
+type SettingsFileType int
+
+const (
+	// GlobalSettingsFile ...
+	GlobalSettingsFile SettingsFileType = iota
+	// ProjectSettingsFile ...
+	ProjectSettingsFile SettingsFileType = iota
+)
+
+// GetSettingsFile ...
+func GetSettingsFile(settingsFileType SettingsFileType, src string, fileUtils piperutils.FileUtils, httpClient piperhttp.Downloader) error {
+
+	var dest string
+	var err error
+
+	switch settingsFileType {
+	case GlobalSettingsFile:
+		dest, err = getGlobalSettingsFileDest()
+	case ProjectSettingsFile:
+		dest, err = getProjectSettingsFileDest()
+	default:
+		return errors.New("Invalid SettingsFileType")
+	}
+
+	if err != nil {
+		return err
+	}
+
+	if len(src) > 0 {
```

Comment:
For the sake of happy path I would invert the condition

Reply:
changed